openmc/mgxs/library.py

```diff
@@ -717,3 +721,292 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+            nuclide this will be set to 'macro' regardless.
+        domain_names : Iterable of str
+            List of names to apply to the "xsdata" entries in the
+            resultant mgxs data file. Defaults to 'set1', 'set2', ...
+        xs_ids : str or Iterable of str
+            Cross section set identifier (i.e., '71c') for all
+            data sets (if only str) or for each individual one
+            (if iterable of str). Defaults to '1m'.
+
+        Returns
+        -------
+        mgxs_file : openmc.MGXSLibrary
+            Multi-Group Cross Section File that is ready to be printed to the
+            file of choice by the user.
+
+        Raises
+        ------
+        ValueError
+            When the Library object is initialized with insufficient types of
+            cross sections for the Library.
+
+        See also
+        --------
+        Library.dump_to_file(mgxs_lib, filename, directory)
+
+        """
+
+        # Check to ensure the Library contains the correct
+        # multi-group cross section types
+        self.check_library_for_openmc_mgxs()
+
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+        if domain_names is not None:
+            cv.check_iterable_type('domain_names', domain_names, basestring)
+        if xs_ids is not None:
+            if isinstance(xs_ids, basestring):
+                # If we only have a string lets convert it now to a list
+                # of strings.
+                xs_ids = [xs_ids for i in range(len(self.domains))]
+            else:
+                cv.check_iterable_type('xs_ids', xs_ids, basestring)
+        else:
+            xs_ids = ['1m' for i in range(len(self.domains))]
+
+        # If gathering material-specific data, set the xs_type to macro
+        if not self.by_nuclide:
+            xs_type = 'macro'
+
+        # Initialize file
+        mgxs_file = openmc.MGXSLibrary(self.energy_groups)
+
+        # Set the scattering order as isotropic until
+        # support for higher orders are included in openmc.mgxs
+        order = 0
+
+        # Build storage for our XSdata objects
+        xsdatas = []
+
+        for i, domain in enumerate(self.domains):
+            if self.by_nuclide:
+                nuclides = list(domain.get_all_nuclides().keys())
+            else:
+                nuclides = ['total']
+            for nuclide in nuclides:
+                # Build & add metadata to XSdata object
+                if domain_names is None:
+                    name = 'set' + str(i + 1)
+                else:
+                    name = domain_names[i]
+                if nuclide is not 'total':
+                    name += '_' + nuclide
+
+                xsdata = self.get_xsdata(domain, name, nuclide=nuclide,
+                                         xs_type=xs_type, xs_id=xs_ids[i],
+                                         order=order)
+
+                xsdatas.append(xsdata)
```

Comment:
Could just use `mgxs_file.add_xsdata()` here and avoid the need to create an xsdatas list.